File: cotta-guice/src/main/java/com/mgtriffid/games/cotta/experimental/guice/serialization/maps/recipe/MapComponentRecipe.java
package com.mgtriffid.games.cotta.experimental.guice.serialization.maps.recipe;

import com.mgtriffid.games.cotta.experimental.guice.data.ComponentKey;
import com.mgtriffid.games.cotta.experimental.guice.serialization.recipe.ComponentRecipe;

import java.util.Map;

public class MapComponentRecipe implements ComponentRecipe {
    public final ComponentKey componentKey;
    public final Map<String, Object> data;

    public MapComponentRecipe(ComponentKey componentKey, Map<String, Object> data) {
        this.componentKey = componentKey;
        this.data = data;
    }
}

File: cotta-guice/src/main/kotlin/com/mgtriffid/games/cotta/experimental/guice/serialization/impl/maps/recipe/MapComponentRecipe.kt
package com.mgtriffid.games.cotta.experimental.guice.serialization.impl.maps.recipe

import com.mgtriffid.games.cotta.experimental.guice.data.ComponentKey
import com.mgtriffid.games.cotta.experimental.guice.serialization.recipe.ComponentRecipe

class MapComponentRecipe(
    val componentKey: ComponentKey.StringComponentKey, val data: Map<String, Any>
) : ComponentRecipe
